### DISCUSSION TITLE
Give me a solution for the esp01 response throught Micropython

### DISCUSSION THREAD

**Discussion by Jayakrishna112** [micropython/micropython#13762] [Hardware & Peripherals]
### Checks

- [X] I agree to follow the MicroPython [Code of Conduct](https://github.com/micropython/micropython/blob/master/CODEOFCONDUCT.md) to ensure a safe and respectful space for everyone.

- [X] I've searched for [existing issues](https://github.com/micropython/micropython/issues) matching this bug, and didn't find any.

### Port, board and/or hardware

Raspberry Pi Pico

### MicroPython version

MicroPython v1.22.1 on 2024-01-05; Raspberry Pi Pico with RP2040

### Reproduction

when i run this python code :

```py
from machine import UART, Pin
import utime

esp = UART(0, baudrate=115200)

def send_at_command(cmd):
    esp.write(cmd)
    response = waitResp(esp)
    # print(response.decode())
    return response

def waitResp(uart, timeout=2000):
    prvMills = utime.ticks_ms()
    resp = b""
    while (utime.ticks_ms()-prvMills) < timeout:
        if uart.any():
            resp = b"".join([resp, uart.readline()])
    print("resp:")
    try:
        print(resp.decode())
    except UnicodeError:
        print("E:", resp)
    return resp

send_at_command(b'AT')
```

### Expected behaviour

the output should be:

```
resp:
AT

OK
```

### Observed behaviour

but giving only:

```
resp:
At
```

### Additional Information

No, I've provided everything above.

**Comment by projectgus** [comment]
Hi @Jayakrishna112 ,

We converted this from an Issue to a Discussion as it's not a bug in MicroPython.

AT commands need to be ended with a newlines (return) character, either `\r\n` or `\n`. So try changing your function like this:

```py
def send_at_command(cmd):
    esp.write(cmd)
    esp.write('\r\n')  # New line
   ... rest of function stays same
```

**Comment by Jayakrishna112** [comment]
It has worked
